### PR TITLE
Unit test fix setup_chains()

### PR DESCRIPTION
**Pull request by dsiganos**
Fixing a race condition in unit test function setup_chains()

### CODE REVIEW

**Review by thsfs — DISMISSED** [review]

File: nano/test_common/testutil.hpp [hunk 1]
```diff
@@ -56,13 +56,15 @@
...
-#define EXPECT_TIMELY(time, condition)             \
-	system.deadline_set (time);                    \
-	std::error_code ec;                            \
-	while (!(condition) && !(ec = system.poll ())) \
-	{                                              \
-	}                                              \
-	EXPECT_NO_ERROR (ec);
+#define EXPECT_TIMELY(time, condition)                  \
+	system.deadline_set (time);                         \
+	{                                                   \
```
Comment (clemahieu) [inline nano/test_common/testutil.hpp:61]: Whats's the reason for the extra scope here?

**Review by pwojcikdev — DISMISSED** [review]

**Review by clemahieu — DISMISSED** [review]

File: nano/test_common/testutil.hpp (hunk 1 above)
Comment (dsiganos) [reply nano/test_common/testutil.hpp:61]: It is a C style macro that creates a local variable.
It is not ideal, so I limited its scope and added an underscore to the variable name.

**Comment by pwojcikdev** [comment]
Every call to `nano::test::confirm()` should be done with ASSERT_TIMELY. I implemented that helper with intention to be used that way, I'm not sure why I didn't do that in other places.

**Comment by dsiganos** [comment]
Hmmm, OK, I can see more calls to nano::test::confirm()  that do not use TIMELY asserts, I'll review/fix them.

**Pull request merged**